compiler/rustc_const_eval/src/interpret/intern.rs

```diff
@@ -33,7 +37,7 @@ pub trait CompileTimeMachine<'mir, 'tcx: 'mir, T> = Machine<
         FrameExtra = (),
         AllocExtra = (),
         MemoryMap = FxIndexMap<AllocId, (MemoryKind<T>, Allocation)>,
-    >;
+    > + const_eval::HasStaticRootDefId;
```

Comment:
Since this is where the trait is needed, IMO it'd make more sense to declare it here.